@barefootjs/test: TS2366 in ir-to-test-node.ts convert()

## Problem

`bun run --filter @barefootjs/test build` fails with:

\`\`\`
src/ir-to-test-node.ts(30,60): error TS2366: Function lacks ending return statement and return type does not include 'undefined'.
\`\`\`

This is a pre-existing error (not introduced by any recent refactor — reproducible on \`main\` prior to #891).

## Location

\`packages/test/src/ir-to-test-node.ts:30\` — \`convert(node, cmap): TestNode\`

The \`switch (node.type)\` covers the known IR node types but TypeScript cannot prove exhaustiveness, so when no branch matches the function implicitly falls through to \`undefined\`, which is not in the declared return type.

## Suggested fix

One of:
- Add a \`default: throw new Error(\`Unhandled IR node type: \${(node as IRNode).type}\`)\` branch
- Change the return type to \`TestNode | undefined\` if some node types are intentionally unhandled

## Context

Surfaced during #891 (packages/client-runtime merge). Not blocking that PR because it was already failing before.

--- a/packages/test/__tests__/render-to-test.test.ts
+++ b/packages/test/__tests__/render-to-test.test.ts
@@ -125,6 +125,30 @@ export { Static }
   })
 })
 
+describe('<Async> streaming boundary', () => {
+  test('exposes resolved children as a fragment', () => {
+    const source = `
+function ProductPage() {
+  return (
+    <div>
+      <Async fallback={<p>Loading...</p>}>
+        <span>Resolved</span>
+      </Async>
+    </div>
+  )
+}
+
+export { ProductPage }
+`
+    const result = renderToTest(source, 'product-page.tsx')
+    const span = result.find({ tag: 'span' })
+    expect(span).not.toBeNull()
+    expect(span!.text).toBeNull()
+    const textChild = span!.children.find(c => c.type === 'text')
+    expect(textChild?.text).toBe('Resolved')
+  })
+})
+
 describe('Error detection', () => {
   test('missing "use client" reports BF001', () => {
     const source = `

--- a/packages/test/src/ir-to-test-node.ts
+++ b/packages/test/src/ir-to-test-node.ts
@@ -16,6 +16,7 @@ import type {
   IRSlot,
   IRIfStatement,
   IRProvider,
+  IRAsync,
 } from '@barefootjs/jsx'
 
 type IRAttribute = IRElement['attrs'][number]
@@ -49,6 +50,12 @@ function convert(node: IRNode, cmap: Map<string, string>): TestNode {
       return convertIfStatement(node, cmap)
     case 'provider':
       return convertProvider(node, cmap)
+    case 'async':
+      return convertAsync(node, cmap)
+    default: {
+      const _exhaustive: never = node
+      throw new Error(`Unhandled IR node type: ${(_exhaustive as IRNode).type}`)
+    }
   }
 }
 
@@ -317,6 +324,30 @@ function convertProvider(node: IRProvider, cmap: Map<string, string>): TestNode 
   const children = node.children.map(c => convert(c, cmap))
 
   if (children.length === 1) return children[0]
+
+  return new TestNode({
+    tag: null,
+    type: 'fragment',
+    children,
+    text: null,
+    props: {},
+    classes: [],
+    role: null,
+    aria: {},
+    dataState: null,
+    events: [],
+    reactive: false,
+    componentName: null,
+  })
+}
+
+// ---------------------------------------------------------------------------
+// Async
+// ---------------------------------------------------------------------------
+
+function convertAsync(node: IRAsync, cmap: Map<string, string>): TestNode {
+  // Represent the resolved content as a fragment; tests assert on final state.
+  const children = node.children.map(c => convert(c, cmap))
 
   return new TestNode({
     tag: null,

